lib/core/error/error-with-props.ts

```diff
@@ -0,0 +1,33 @@
+import { isPlainObject } from "@lib/core/typeguards";
+
+type Params = {
+  message?: string;
+  props?: Record<string, unknown> | unknown;
+};
+
+export class ErrorWithProps extends Error {
```

Comment:
I don't see this used nor the point of it? We should never throw anything anywhere that isn't an `Error`